### PR TITLE
fix: WhatsApp reply forwarding, free model filtering & message footer UI

### LINKED ISSUE #636: WhatsApp: Forward platform replies to WhatsApp user
## Problem

Messages typed in the Synaplan web UI within a WhatsApp-sourced chat are stored locally but never forwarded to the WhatsApp user's phone.

---

## Expected

When a platform user replies in a WhatsApp chat, the message should be delivered to the WhatsApp user via the Meta Graph API.

## Actual

The reply is saved as a `WEB` message in the database only. The WhatsApp user never receives it. No error or warning is shown to the platform user.

---

## Steps to reproduce

1. A WhatsApp user sends a message to the Synaplan business number
2. The message appears as a WhatsApp chat in the Synaplan platform
3. A platform user opens the chat and types a reply
4. The reply appears in the Synaplan UI but does not arrive on the WhatsApp user's phone

---

## Notes

- `StreamController` and `MessageController` always create messages with `providerIndex('WEB')` / `messageType('WEB')` regardless of `Chat.source`
- `WhatsAppService` is not injected in either controller — only used in `WebhookController` and `PhoneVerificationController`
- Automatic AI replies to WhatsApp messages work correctly (sent via `WhatsAppService::sendMessage`)
- Phone number metadata needed for sending is already stored on inbound messages

### PR DESCRIPTION
**Pull request by cristiangrxs**
## Summary

Forward AI responses to WhatsApp users when replies originate from the web UI, hide free models (price = 0) from selection dropdowns with admin override, and fix message footer truncation.

## Changes
- **WhatsApp forwarding (closes #636):** New `MessageForwardingService` that forwards AI responses to WhatsApp when `Chat.source === 'whatsapp'`. Retrieves phone number and phone number ID from inbound message metadata. Best-effort: failures are logged but never block the web UI stream. Integrated into `StreamController` after the outgoing message is persisted.
- **Free model filtering:** Models with `priceIn = 0` AND `priceOut = 0` are now excluded from `GET /api/v1/config/models`, affecting all model dropdowns (chat input, task prompts, widget settings, regenerate). New `BSHOWWHENFREE` column on `BMODELS` (default 0) allows admins to override this per model via a "Free OK" toggle in the admin panel.
- **Message footer UI:** Model name in the footer badge (`Chat Model: xyz`) now truncates with ellipsis instead of pushing the timestamp to a second line. Label ("Chat Model:") and icon remain always visible via `flex-shrink-0`.

## Verification
- [ ] Manual
- [ ] Tests added/updated

## Notes

- Closes #636 — WhatsApp: Forward platform replies to WhatsApp user
- Migration required: `BSHOWWHENFREE` column added to `BMODELS` table
- The free-model filter operates at API level, so all existing and future model dropdowns are covered automatically

### CODE REVIEW

File: frontend/src/i18n/en.json [hunk 1]
```diff
@@ -772,6 +772,10 @@
         "unitIn": "Unit in",
         "priceOut": "Price out",
         "unitOut": "Unit out",
+        "showWhenFree": "Free OK",
+        "showWhenFreeTooltip": "Show model in selection even when price is 0 (free)",
+        "showWhenFreeOn": "Override active: model visible despite free pricing",
+        "showWhenFreeOff": "Default: model hidden when pricing is 0",
```
[the remark this answers is not in the document]
Comment (cristiangrxs) [reply frontend/src/i18n/en.json]: The es.json and tr.json files do not have any admin model translations at all (no `editModels`, `priceIn`, `priceOut`, etc.) — they rely entirely on English fallback for the admin section. Adding only `showWhenFree` keys without the full admin block would be inconsistent. This is an existing pattern, not a new gap.

[the remark this answers is not in the document]
Comment (cristiangrxs) [reply frontend/src/i18n/en.json]: Fixed — updated tooltip strings to mention both priceIn and priceOut in 1ce024d.

File: frontend/src/i18n/de.json [hunk 2]
```diff
@@ -889,6 +889,10 @@
         "unitIn": "Einheit ein",
         "priceOut": "Preis aus",
         "unitOut": "Einheit aus",
+        "showWhenFree": "Gratis OK",
+        "showWhenFreeTooltip": "Modell in der Auswahl anzeigen, auch wenn der Preis 0 ist (kostenlos)",
+        "showWhenFreeOn": "Override aktiv: Modell trotz kostenlosem Preis sichtbar",
+        "showWhenFreeOff": "Standard: Modell bei Preis 0 ausgeblendet",
```
[the remark this answers is not in the document]
Comment (cristiangrxs) [reply frontend/src/i18n/de.json]: Fixed — German tooltips updated to mention both prices in 1ce024d.

File: frontend/src/components/ChatMessage.vue [hunk 3]
```diff
@@ -528,28 +528,32 @@
...
                 :title="getModelTypeTitle"
                 data-testid="btn-message-model-chat"
                 @click="showModelDetails('chat')"
               >
-                <Icon :icon="getModelTypeIcon" class="w-3.5 h-3.5" />
-                <span class="hidden sm:inline">{{ getModelTypeLabel }}:</span>
-                <span class="font-semibold">{{ shortenModel(aiModels.chat.model) }}</span>
+                <Icon :icon="getModelTypeIcon" class="w-3.5 h-3.5 flex-shrink-0" />
+                <span class="hidden sm:inline flex-shrink-0">{{ getModelTypeLabel }}:</span>
+                <span class="font-semibold truncate">{{ shortenModel(aiModels.chat.model) }}</span>
```
[the remark this answers is not in the document]
Comment (cristiangrxs) [reply frontend/src/components/ChatMessage.vue]: Fixed — added `min-w-0` to both model-name spans in 1ce024d.

File: backend/src/Service/Message/MessageForwardingService.php [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+                    'to' => $toPhone,
+                    'error' => $result['error'] ?? 'Unknown error',
+                ]);
+            }
+        } catch (\Exception $e) {
+            $this->logger->error('Exception while forwarding message to WhatsApp', [
+                'chat_id' => $chatId,
+                'to' => $toPhone,
+                'error' => $e->getMessage(),
+            ]);
```
[the remark this answers is not in the document]
Comment (cristiangrxs) [reply backend/src/Service/Message/MessageForwardingService.php]: Good catch — changed from `\Exception` to `\Throwable` in 9860984d.

File: backend/src/Controller/StreamController.php [hunk 5]
```diff
@@ -1043,6 +1045,10 @@ function ($statusUpdate) {
 
                 $this->em->flush();
 
+                if (!$isWidgetMode) {
+                    $this->messageForwardingService->forwardIfNeeded($chat, $finalText);
```
[the remark this answers is not in the document]
Comment (cristiangrxs) [reply backend/src/Controller/StreamController.php:1049]: The PCNTL alarm approach is fragile and not portable. The proper solution would be dispatching via Symfony Messenger queue, but that is out of scope for this PR. The current implementation is already best-effort (failures are logged, never propagated). Added `\Throwable` catch for robustness.

File: backend/src/Repository/MessageRepository.php [hunk 6]
```diff
@@ -305,6 +305,29 @@ public function getLastMessageTextForChats(array $chatIds): array
...
+            ->andWhere('m.direction = :direction')
+            ->andWhere('channelMeta.metaValue = :channel')
+            ->setParameter('channelKey', 'channel')
+            ->setParameter('chatId', $chatId)
+            ->setParameter('direction', 'IN')
+            ->setParameter('channel', $channel)
+            ->orderBy('m.id', 'DESC')
+            ->setMaxResults(1)
+            ->getQuery()
+            ->getOneOrNullResult();
```
[the remark this answers is not in the document]
Comment (cristiangrxs) [reply backend/src/Repository/MessageRepository.php:328]: MessageForwardingService unit tests cover this method indirectly via mock expectations (verifying `findLatestInboundByChannel` is called with correct args). Adding a full integration test for the repository method requires KernelTestCase with DB fixtures, which is a separate concern.

File: backend/src/Entity/Model.php [hunk 7]
```diff
@@ -55,6 +55,10 @@ class Model
     #[ORM\Column(name: 'BACTIVE', type: 'integer', options: ['default' => 1])]
     private int $active = 1;
 
+    /** @var int Override to show this model in selection even when both prices are 0 */
+    #[ORM\Column(name: 'BSHOWWHENFREE', type: 'integer', options: ['default' => 0])]
+    private int $showWhenFree = 0;
+
```
[the remark this answers is not in the document]
Comment (cristiangrxs) [reply backend/src/Entity/Model.php:61]: Migration already exists: `Version20260331063312.php` adds `BSHOWWHENFREE INT DEFAULT 0 NOT NULL` to `BMODELS`.

File: backend/src/Controller/ConfigController.php [hunk 8]
```diff
@@ -384,9 +384,12 @@ public function getModels(#[CurrentUser] ?User $user): JsonResponse
             ['quality' => 'DESC', 'rating' => 'DESC']
         );
 
-        // Build model list with tag information
+        // Build model list with tag information, excluding free models without override
         $modelList = [];
         foreach ($models as $model) {
+            if ($model->isHiddenBecauseFree()) {
+                continue;
+            }
```
[the remark this answers is not in the document]
Comment (cristiangrxs) [reply backend/src/Controller/ConfigController.php:392]: Added `ModelEntityTest` with 6 tests covering `isHiddenBecauseFree()` in all edge cases. The filtering in ConfigController delegates to this entity method, so it is fully covered.

File: frontend/src/i18n/en.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (cristiangrxs) [reply frontend/src/i18n/en.json]: es.json and tr.json have no admin model translations at all — they rely on English fallback for the entire admin section. Adding only showWhenFree without the full block would be inconsistent.

[the remark this answers is not in the document]
Comment (cristiangrxs) [reply frontend/src/i18n/en.json]: Fixed — updated tooltip strings to mention both priceIn and priceOut in 1ce024d.

File: frontend/src/i18n/de.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (cristiangrxs) [reply frontend/src/i18n/de.json]: Fixed — German tooltips updated to mention both prices in 1ce024d.

File: frontend/src/components/ChatMessage.vue (hunk 3 above)
[the remark this answers is not in the document]
Comment (cristiangrxs) [reply frontend/src/components/ChatMessage.vue]: Fixed — added min-w-0 to both model-name spans in 1ce024d.

File: backend/src/Service/Message/MessageForwardingService.php (hunk 4 above)
[the remark this answers is not in the document]
Comment (cristiangrxs) [reply backend/src/Service/Message/MessageForwardingService.php]: Good catch — changed from Exception to Throwable in 9860984d.

File: backend/src/Controller/StreamController.php (hunk 5 above)
[the remark this answers is not in the document]
Comment (cristiangrxs) [reply backend/src/Controller/StreamController.php:1049]: PCNTL alarm approach is fragile and not portable. Proper solution would be Symfony Messenger queue (out of scope). Current impl is best-effort with Throwable catch.

File: backend/src/Repository/MessageRepository.php (hunk 6 above)
[the remark this answers is not in the document]
Comment (cristiangrxs) [reply backend/src/Repository/MessageRepository.php:328]: MessageForwardingService unit tests cover this method indirectly via mock expectations. Full DB integration test is a separate concern.

File: backend/src/Entity/Model.php (hunk 7 above)
[the remark this answers is not in the document]
Comment (cristiangrxs) [reply backend/src/Entity/Model.php:61]: Migration already exists: Version20260331063312.php adds BSHOWWHENFREE column.

File: backend/src/Controller/ConfigController.php (hunk 8 above)
[the remark this answers is not in the document]
Comment (cristiangrxs) [reply backend/src/Controller/ConfigController.php:392]: Added ModelEntityTest with 6 tests covering isHiddenBecauseFree() in all edge cases.

File: backend/src/Service/Message/MessageForwardingService.php [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+     * so the web UI flow is not disrupted.
+     */
+    public function forwardIfNeeded(Chat $chat, string $text): void
+    {
+        if ('whatsapp' !== $chat->getSource()) {
+            return;
+        }
+
+        $this->forwardToWhatsApp($chat, $text);
+    }
```
[the remark this answers is not in the document]
Comment (cristiangrxs) [reply backend/src/Service/Message/MessageForwardingService.php:41]: Added MessageForwardingServiceTest with 7 unit tests covering all branches.
